share/pot/set-status.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/sh
+# shellcheck disable=SC3033,SC3040,SC3043
+:
+
+: "${_POT_INTERNAL_STATUS:="starting started stopping stopped"}"
+
+set-status-help()
+{
+	cat <<-"EOH"
+	Internal command, DO NOT USE IF YOU DON'T KNOW WHAT YOU ARE DOING!
+	This command is meant to be invoked using lockf
+	pot set-status [-hv] [-p pname] [-s status]
+	  -h print this help
+	  -v verbose
+	  -p pname : pot name
+	  -s status : the status [starting started stopping stopped]
+	EOH
+}
+
+# $1 pot name
+_get_status()
+{
+	local _pname _status_file
+	_pname="$1"
+	_status_file="${POT_TMP:-/tmp}/pot_status_${_pname}"
+
+	if [ ! -e "$_status_file" ]; then
+		return
+	fi
+	_value="$(grep "^pot.status=" "$_status_file" | tr -d ' \t"' | cut -f2 -d'=' )"
+	echo "$_value"
+}
+
+# $1 pot name
+# $2 new status
+_set_status()
+{
+	local _pname _status_file _new_status
+	_pname="$1"
+	_new_statu="$2"
+	_status_file="${POT_TMP:-/tmp}/pot_status_${_pname}"
+
+	${SED} -i '' -e "/^pot.status=.*/d" "$_status_file"
+	echo "pot.status=$_new_status" >> "$_status_file"
```

Comment:
The expression should read `-e "/^pot\.status=/d"` (dot needs to be escaped, `.*` not required).

You could change this logic, so it's more atomic (only in combination with `tail -n1` above):

```suggestion
	echo "pot.status=$_new_status" >> "$_status_file"
	# remove first (and outdated) occurrence of pot.status
	${SED} -i '' -n -e ":a" \
	  -e '/^pot\.status=/{n;bc' -e ':c' -e 'p;n;bc' -e '}' \
	  -e "p;n;ba" "$_status_file"
```

Reply:
In case it isn't apparent at first glance, this is how the sed command works:

```perl
# pseudo-code                            # sed

while getline() {                        # a:                    (label a)
    if line.starts_with("pot.status=") { #   /^pot\.status=/{
        next_line();                     #     n
        while getline() {                #     c:                (label c)
            println(line);               #       p
            next_line();                 #       n
        }                                #     bc                (goto c)
        break;                           #     (implicit/eof)
    }                                    #   }
    println(line);                       #   p
    next_line();                         #   n
}                                        # ba                    (goto a)
```

Reply:
This is brilliant!